ipmininet/examples/exabgp_prefix_injector.py

```diff
@@ -0,0 +1,126 @@
+import ipaddress
+import itertools
+import random
+
+from ipmininet.iptopo import IPTopo
+from ipmininet.router.config import ebgp_session, RouterConfig, BGP, ExaBGPDaemon, BGPRoute, BGPAttribute, \
+    ExaList
+import ipmininet.router.config.bgp as _bgp
+
+__MAX_UINT32_ = 4294967295
+
+
+def list_of_rnd_lists(nb, max_sub_rnd_list):
+    def random_gen(low, high):
+        while True:
+            yield random.randrange(low, high)
+
+    rnd_lists = []
+
+    for _ in range(0, nb):
+        rnd_set = set()
+        gen = random_gen(1, 65536)
+        rnd_path_len = random.randint(1, max_sub_rnd_list)
+
+        # Try to add elem to set until set length is less than 'rnd_path_len'
+        for x in itertools.takewhile(lambda y: len(rnd_set) <= rnd_path_len, gen):
+            rnd_set.add(x)
+
+        rnd_lists.append(list(rnd_set))
+
+    return rnd_lists
+
+
+def rnd_list(max_sub_rnd_list, strict=False):
+    def random_gen(low, high):
+        while True:
+            yield random.randrange(low, high)
+
+    rnd_set = set()
+    gen = random_gen(1, 65536)
+    rnd_path_len = random.randint(1, max_sub_rnd_list) if not strict else max_sub_rnd_list
+
+    for x in itertools.takewhile(lambda y: len(rnd_set) <= rnd_path_len, gen):
+        rnd_set.add(x)
+
+    return list(rnd_set)
+
+
+def build_bgp_route(ip_networks, my_as):
+
+    my_routes = list()
+
+    for ip_network in ip_networks:
+        next_hop = BGPAttribute("next-hop", "self")
+        as_path = BGPAttribute("as-path", ExaList([my_as] + rnd_list(random.randint(1, 25))))
+        communities = BGPAttribute("community",
+                                   ExaList(["%d:%d" % (j, k) for j, k in zip(rnd_list(24, True), rnd_list(24, True))]))
+        med = BGPAttribute("med", random.randint(1, __MAX_UINT32_))
+        origin = BGPAttribute("origin", random.choice(["igp", "egp", "incomplete"]))
+
+        my_routes.append(BGPRoute(ip_network, [next_hop, origin, med, as_path, communities]))
+
+    return my_routes
+
+
+class ExaBGPTopoInjectPrefixes(IPTopo):
```

Comment:
Also, I forgot but could you add a test that checks your example as well in ```tests/test_hexabgp.py``` ?

Reply:
I refactor the tests from 2c5c17d70344122330a1cd74ad55de669f554fad to fb6a877e58a430e3ee1ab416dfb0dc691a14fd3f